data_structures/src/error.rs

```diff
@@ -200,6 +210,44 @@ pub enum TransactionError {
         expected: Vec<PublicKeyHash>,
         found: Vec<PublicKeyHash>,
     },
+    /// Invalid collateral in data request
+    #[fail(
+        display = "The specified collateral ({} nwits), is less than the minimum required ({} nwits)",
+        value, min
+    )]
+    InvalidCollateral { value: u64, min: u64 },
+    /// Negative collateral in commit transaction
+    #[fail(
+        display = "Tried to create value in commit collateral. Input value: {}, output value: {}",
+        input_value, output_value
+    )]
```

Comment:
This error description makes it look like it was an error that happened _while trying to create value_ (which obviously makes no sense hehehe). It could be something more in the line of describing what's exactly what we are preventing from happening. What is it in this case?

Reply:
This is when a `CommitTransaction` tries to use a negative collateral (more value in outputs than in inputs). Will change to

    Negative collateral in commit transaction. Input value: {}, output value: {}